pyanaconda/modules/base_interface.py

```diff
@@ -0,0 +1,104 @@
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from gi.overrides.GLib import Variant as GVariant
+from pykickstart.errors import KickstartError
+
+from pyanaconda.dbus.template import InterfaceTemplate
+from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
+from pyanaconda.dbus.interface import dbus_interface
+from pyanaconda.dbus.constants import DBUS_MODULE_NAMESPACE
+
+
+@dbus_interface(DBUS_MODULE_NAMESPACE)
+class KickstartModuleInterface(InterfaceTemplate):
+    """DBus interface of a kickstart module.
+
+    The implementation is provided by the KickstartModule class.
+    """
+
+    def AvailableTasks(self) -> List[Tuple[Str, Str]]:
+        """Return DBus object paths for tasks available for this module.
+
+        :returns: List of tuples (Name, DBus object path) for all Tasks.
+                  See pyanaconda.task.Task for Task API.
+        """
+        result = []
+
+        for task in self.implementation.published_tasks:
+            result.append((task.Name, task.object_path))
+
+        return result
+
+    @property
+    def KickstartCommands(self) -> List[Str]:
+        """Return names of kickstart commands handled by module.
+
+        :returns: List of names of kickstart commands handled by module.
+        """
+        return self.implementation.kickstart_command_names
+
+    @property
+    def KickstartSection(self) -> List[Str]:
+        """Return names of kickstart sections handled by module.
+
+        :returns: List of names of kickstart sections handled by module.
+        """
+        return self.implementation.kickstart_section_names
+
+    @property
+    def KickstartAddons(self) -> List[Str]:
+        """Return names of kickstart addons handled by module.
+
+        :returns: List of names of kickstart addons handled by module.
+        """
+        return self.implementation.kickstart_addon_names
+
+    def ReadKickstart(self, kickstart: Str) -> Dict[Str, Variant]:
+        """Read the kickstart string.
+
+        :param kickstart: a kickstart string
+        :returns: a dictionary with a result
+        """
+        try:
+            self.implementation.read_kickstart(kickstart)
+        except KickstartError as e:
+            return {
+                "success": GVariant('b', False),
+                "error_message": GVariant('s', str(e)),
+                "error_lineno": GVariant('i', 1)
```

Comment:
`lineno` will be always 1? If this is only a placeholder than please comment or use something else.

Reply:
Fixed.